ports/mimxrt/adc.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2021 Philipp Ebensberger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef MICROPY_INCLUDED_MIMXRT_ADC_H
+#define MICROPY_INCLUDED_MIMXRT_ADC_H
+
+#include <stdint.h>
+#include "py/obj.h"
+#include "fsl_adc.h"
+
+
+typedef struct _machine_adc_obj_t {
+    mp_obj_base_t base;
+    ADC_Type *adc;
+    uint8_t channel;
+    uint8_t channel_group;
+    uint16_t resolution;
+} machine_adc_obj_t;
+
+extern const mp_obj_type_t machine_adc_type;
```

Comment:
This type and the typedef just above it really belong in `modmachine.h`.

Reply:
For now all of the machine modules have that structure in this port. Would it be fine with you if I keep the structure for now but refactor all of the modules (led, pin, adc) in a separate PR?

Reply:
The only existing header file is `modmachine.h`, which contains the types for timer and rtc.  So would be good to put at least `machine_adc_type` in `modmachine.h`.  The rest can be changed later.

Reply:
Ok sure - I can do that